### PR TITLE
Add GCP Broker provider tool

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

Add scripts which:
- installs/uninstalls GCP Service Broker in a given namespace
- checks if GCP Broker is registered in Service Catalog

##### Example output from checker script: 
```bash

kubectl version
Client Version: v1.10.0
Server Version: v1.10.7

Checking if GCP ServiceBroker is registered in Service Catalog...
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 1/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 2/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 3/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 4/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 5/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 6/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 7/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 8/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 9/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 10/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 11/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 12/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 13/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 14/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 15/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
Error from server (NotFound): servicebrokers.servicecatalog.k8s.io "gcp-broker" not found
[Retry 16/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
[Retry 17/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
[Retry 18/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
[Retry 19/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
[Retry 20/40] GCP ServiceBroker condition Ready is not set to True - retry in 3s
GCP ServiceBroker is registered successfully.
```

##### Example output from installer script:
```bash

kubectl version
Client Version: v1.10.0
Server Version: v1.10.7

gcloud version
Google Cloud SDK 225.0.0
bq 2.0.37
core 2018.11.09
gsutil 4.34

sc version
sc version 0.1.1 linux/amd64

Activated service account credentials for: [adamw-test-account@kyma-project.iam.gserviceaccount.com]
using project:  kyma-project
enabling a GCP API: servicebroker.googleapis.com
enabling a GCP API: bigquery-json.googleapis.com
enabling a GCP API: bigtableadmin.googleapis.com
enabling a GCP API: ml.googleapis.com
enabling a GCP API: pubsub.googleapis.com
enabling a GCP API: spanner.googleapis.com
enabling a GCP API: sqladmin.googleapis.com
enabling a GCP API: storage-api.googleapis.com
enabled required APIs:
  servicebroker.googleapis.com
  bigquery-json.googleapis.com
  bigtableadmin.googleapis.com
  ml.googleapis.com
  pubsub.googleapis.com
  spanner.googleapis.com
  sqladmin.googleapis.com
  storage-api.googleapis.com
generated the key at:  /tmp/service-catalog-gcp983136652/key.json
Broker "default", already exists
Creating k8s resource using 'namespace' template
Creating k8s resource using 'google-oauth-deployment' template
Creating k8s resource using 'service-account-secret' template
Creating k8s resource using 'google-oauth-rbac' template
Creating k8s resource using 'google-oauth-service-account' template
Creating k8s resource using 'gcp-broker' template
The Service Broker has been added successfully.
````

> **Note:** 
We are creating Jobs in a hook, and we cannot rely upon `helm delete` to remove the resources.
To destroy such resources, we added "helm.sh/hook-delete-policy" annotation to the hook template file but only for `hook-succeeded`.
When hooks are failed then we are not removing it automatically, to be able to see its logs. Because of that, we are removing manually those Jobs because Tiller won't do that even on `helm delete` action.
>
> source: https://github.com/helm/helm/blob/master/docs/charts_hooks.md#hook-resources-are-not-managed-with-corresponding-releases

**Related issue(s)**
- https://github.com/kyma-project/kyma/issues/786

### CODE REVIEW

**Comment by mszostok** [comment]
TODOs:

- [ ] define readme file https://github.com/kyma-project/community/blob/master/guidelines/internal-guidelines/repository-template/template/README.md
- [ ] create Jenkinsfile similar to this one https://github.com/kyma-project/kyma/blob/master/tools/etcd-tls-setup/Jenkinsfile
- [ ] add this project in orchestrator.Jenkinsfile file
- [ ] add this project in cont-integration repo
- [ ] adjust codeowners file

File: tools/gcp-broker-provider/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM google/cloud-sdk:224.0.0-alpine
```
Comment (piotrmiskiewicz) [inline tools/gcp-broker-provider/Dockerfile]: there is a newer 225 version

File: tools/gcp-broker-provider/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+LABEL source="git@github.com:kyma-project/kyma.git"
+
+ARG KUBECTL_CLI_VERSION=1.10.0
```
Comment (piotrmiskiewicz) [inline tools/gcp-broker-provider/Dockerfile]: minor: why no to define like v1.10.0?

Comment (adamwalach) [reply tools/gcp-broker-provider/Dockerfile]: fixed

File: tools/gcp-broker-provider/Dockerfile (hunk 1 above)
Comment (adamwalach) [reply tools/gcp-broker-provider/Dockerfile]: updated

File: tools/gcp-broker-provider/bin/run.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+# Following script provisions and deprovisions GKE cluster.
```
Comment (PK85) [inline tools/gcp-broker-provider/bin/run.sh]: ? fix to adds/remowes GCP Broker

File: tools/gcp-broker-provider/bin/run.sh [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+#
+# Expected environment variables:
```
Comment (PK85) [inline tools/gcp-broker-provider/bin/run.sh]: I assume those envs are needed? If yes, please say something more in README.md:
- Prerequisites - google setup
- Usage: how to run checker.sh and run.sh (I understand those script are executed via bundle/JOb)

File: tools/gcp-broker-provider/bin/gcp-broker.sh [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    if [ "${discoveredUnsetVar}" = true ] ; then
+        exit 1
+    fi
+}
+
+function configureGCloud() {
+    if kubectl get secrets -n ${WORKING_NAMESPACE} | grep -q "${GCP_SECRET_NAME}"; then
+        export GOOGLE_APPLICATION_CREDENTIALS=${CURRENT_DIR}"/sa-key.json"
+
+        projectName=$(kubectl get secret gcp-broker-data -o jsonpath='{ .data.project-name }' | base64 -D)
```
Comment (adamwalach) [inline tools/gcp-broker-provider/bin/gcp-broker.sh]: base64 in linux does not accept "-D", use "-d" instead

Comment (mszostok) [reply tools/gcp-broker-provider/bin/gcp-broker.sh]: FYI: I changed it to full name `--decode` because we want to support both linux and osx

**Pull request merged**